tests/testCheckerMain.ml

```diff
@@ -113,7 +273,205 @@ let suite =
+           * out Cancel_liquidation_slice here, but to do that we need to be able to find
+           * th leaf_ptr of the slice, and currently that's hard to do. If/when we
+           * address #150 we should be able to do that. *)
+
+          (* setup: touch to start the auction *)
+          Ligo.Tezos.new_transaction ~seconds_passed:63 ~blocks_passed:1 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Touch ()))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+          (* Liquidation_auction_place_bid *)
+          let min_bid = CheckerEntrypoints.wrapper_view_current_liquidation_auction_minimum_bid ((), sealed_wrapper) in
+          Ligo.Tezos.new_transaction ~seconds_passed:394 ~blocks_passed:6 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Liquidation_auction_place_bid (min_bid.auction_id, min_bid.minimum_bid)))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+(*
+          (* FIXME: WE HAVE A PROBLEM I THINK: THE FOLLOWING RAISES AN
+           * internalError_KitSubNegative error, which I thought was impossible.
+           * Is that because of how high the index is? Is there a bug somewhere? *)
+          (* setup: make enough time pass so that the auction finishes *)
+          let seconds_passed = int_of_string (Ligo.string_of_int (Ligo.add_int_int Constants.max_bid_interval_in_seconds (Ligo.int_from_literal "1"))) in
+          let blocks_passed = int_of_string (Ligo.string_of_nat (Ligo.add_nat_nat Constants.max_bid_interval_in_blocks (Ligo.nat_from_literal "1n"))) in
+          Ligo.Tezos.new_transaction ~seconds_passed:seconds_passed ~blocks_passed:blocks_passed ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Touch ()))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+*)
+
+          (* Note: to avoid the unused variable warning. *)
+          assert_equal sealed_wrapper sealed_wrapper
+       )
+    );
+
+    ("If checker is sealed, users should be able to call CheckerEntrypoint/LazyParams/Touch_liquidation_slices" >::
+     with_sealed_wrapper
+       (fun sealed_wrapper ->
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Touch_liquidation_slices ([])))) in
+          (* This call should succeed *)
+          Ligo.Tezos.new_transaction ~seconds_passed:0 ~blocks_passed:0 ~sender:alice_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let _ops, _wrapper = CheckerMain.main (op, sealed_wrapper) in
+          ()
+       )
+    );
+
+    (* Note: rather superficial call, to increase path coverage;
+     * I would have liked a call that succeeds here instead, but
+     * #150 makes this annoying to achieve. *)
```

Comment:
:+1: Adding more views would make the end-to-end tests less verbose as well.

Reply:
Yeah, overall I'd like to follow the approach @utdemir had suggested: ideally we'd be able to have testing at such a high level using only the public API (entrypoints + offline views), and not inspecting the storage (much :upside_down_face:). This for both the unit tests at the level of `CheckerMain` and the e2e tests indeed.